src/frontends/pytorch/src/op/as_strided.cpp

```diff
@@ -0,0 +1,106 @@
+        stride_idxs[idx] = {input_strides[idx], idx};
+    });
+
+    std::sort(stride_idxs.begin(), stride_idxs.end(), compare_strides);
+    std::vector<uint64_t> transpose_idx(idxs.size());
+    int transpose_counter = 0;
+    std::for_each(stride_idxs.begin(), stride_idxs.end(), [&](std::tuple<size_t, size_t>& pair) {
+        transpose_idx[transpose_counter] = uint64_t(std::get<1>(pair));
+        transpose_counter++;
+    });
+    auto transpose_idx_const =
+        context.mark_node(v0::Constant::create(element::i32, Shape{transpose_idx.size()}, transpose_idx));
+    auto transposed_input = context.mark_node(std::make_shared<v1::Transpose>(input, transpose_idx_const));
+    auto flat_input = context.mark_node(std::make_shared<v1::Reshape>(transposed_input, const_neg_1, false));
+    std::deque<Output<Node>> sizes;
+    std::deque<Output<Node>> strides;
+    if (std::dynamic_pointer_cast<v0::Constant>(context.get_input_from_visible_context(1).get_node_shared_ptr())) {
+        auto input_vector = context.const_input<std::vector<int64_t>>(1);
+        std::for_each(input_vector.rbegin(), input_vector.rend(), [&](int64_t input_val) {
+            auto const_input = context.mark_node(v0::Constant::create(element::i32, Shape{}, {input_val}));
+            sizes.push_front(const_input);
+        });
+    } else {
+        sizes = get_list_as_outputs(context.get_input(1));
+    }
+    if (std::dynamic_pointer_cast<v0::Constant>(context.get_input_from_visible_context(2).get_node_shared_ptr())) {
+        auto input_vector = context.const_input<std::vector<int64_t>>(2);
+        std::for_each(input_vector.rbegin(), input_vector.rend(), [&](int64_t input_val) {
+            auto const_input = context.mark_node(v0::Constant::create(element::i32, Shape{}, {input_val}));
+            strides.push_front(const_input);
+        });
+    } else {
+        strides = get_list_as_outputs(context.get_input(2));
+    }
+    auto offset = const_0->output(0);
+    if (!context.input_is_none(3)) {
+        offset = context.get_input(3);
+    }
+    FRONT_END_OP_CONVERSION_CHECK(sizes.size() == strides.size(),
+                                  "aten::as_strided: Vector for strides and sizes need to have equal length.");
+    auto strides_size = strides.size() - 1;
+    auto i = 0;
+    auto strides_length_const = context.mark_node(v0::Constant::create(element::i32, Shape{1}, {strides.size()}));
+    auto ones_strides_len = context.mark_node(std::make_shared<v0::Tile>(const_1, strides_length_const));
+    auto indices = const_0;
+    std::for_each(strides.rbegin(), strides.rend(), [&](Output<Node>& stride) {
+        auto const_num_iter = context.mark_node(v0::Constant::create(element::i32, Shape{1}, {strides_size - i}));
+        stride = context.mark_node(std::make_shared<v0::Convert>(stride, element::i32));
+        auto size = context.mark_node(std::make_shared<v0::Convert>(sizes.at(strides_size - i), element::i32));
```

Comment:
No need to convert to same type to use Range
```suggestion
        auto size = sizes.at(strides_size - i);
```